app/extension/services/extension-management-service.ts

```diff
@@ -668,6 +674,32 @@ export class ExtensionManagementService extends Eventable<IExtensionManagementSe
     }
   }
 
+  /**
+   * Create a scheduled task to check for updates daily.
+   */
+  async startUpdateCheckDaemon() {
+    const taskId = "ExtManagementServiceUpdate";
+    try {
+      const task = this._dailyUpdateScheduler.getById(taskId);
+      if (task) {
+        this._dailyUpdateScheduler.removeById(taskId);
+      }
+    } catch (e) {}
+    const task = new Task(
+      taskId,
+      async () => {
+        await this.checkUpdate();
+      },
+      undefined
+    );
+    const job = new SimpleIntervalJob(
+      { days: 1, runImmediately: true },
```

Comment:
Since we call `this.checkUpdate()` in the `initialize()`, we don't need to `runImmediately = true` here?

Reply:
Agreed. Fixed in the latest commit.